cpu/stm32/include/periph/wb/periph_cpu.h

```diff
@@ -32,6 +52,21 @@ extern "C" {
  */
 #define STM32_BOOTLOADER_ADDR   (0x1FFF0000)
 
+/**
+ * @brief   Override ADC resolution values
+ * @{
+ */
+#define HAVE_ADC_RES_T
+typedef enum {
+    ADC_RES_6BIT  = (ADC_CFGR_RES),   /**< ADC resolution: 6 bit */
+    ADC_RES_8BIT  = (ADC_CFGR_RES_1), /**< ADC resolution: 8 bit */
+    ADC_RES_10BIT = (ADC_CFGR_RES_0), /**< ADC resolution: 10 bit */
+    ADC_RES_12BIT = (0x0),            /**< ADC resolution: 12 bit */
+    ADC_RES_14BIT = (0x1),            /**< not applicable */
+    ADC_RES_16BIT = (0x2)             /**< not applicable */
```

Comment:
driver will reject the resolution if the value is 0x03 (dunno why 0x03 though) but it might be a good idea to use this value to reject these resolution and thus prevent misconfiguration.

Reply:
Or maybe update the driver to use these values instead to reject the configuration.

Reply:
> driver will reject the resolution if the value is 0x03 (dunno why 0x03 though) but it might be a good idea to use this value to reject these resolution and thus prevent misconfiguration.

That's not quite accurate, the driver will reject the resolution if the value can be masked by 0x03. This is true for the 0x01 and 0x02 values. In binary it makes more sense: The mask is 0b0011 and the illegal values are 0b0001 and 0b0010.
0b0011 & 0b0001 = 0b0001, 0b0011 & 0b0010 = 0b0010. 
Since the resolution bits RES in the ADC_CFGR register are at position 3 and 4, this can be used to find the two illegal values.

IMO this is not a very pretty solution since it relies on magic numbers and it is not obviously documented. But it seems to be the same for all other ADC implementations as well. For now I wouldn't change it though.

(This part of the code is just copied from the cpu/stm32/include/periph/l4/periph_cpu.h file. It is the same for the L4 and WB55.)

<hr/>

STM32WB55 Reference Manual: https://www.st.com/resource/en/reference_manual/rm0434-multiprotocol-wireless-32bit-mcu-armbased-cortexm4-with-fpu-bluetooth-lowenergy-and-802154-radio-solution-stmicroelectronics.pdf page 489

Reply:
Then you could borrow the behavior from [here](https://github.com/RIOT-OS/RIOT/blob/436b6edb694ecaffc73817c359ac31d65a09524a/cpu/sam0_common/periph/adc.c#L175) which is more easy to understand.

Reply:
> Then you could borrow the behavior from [here](https://github.com/RIOT-OS/RIOT/blob/436b6edb694ecaffc73817c359ac31d65a09524a/cpu/sam0_common/periph/adc.c#L175) which is more easy to understand.

I generally agree with you, but I didn't change anything in the cpu/stm32/periph/adc_l4.c file in this PR. I just renamed it and changed the cpu/stm32/include/wb/periph_cpu.h file to be compatible with the data structures that adc_l4.c expects.


A better approach would be to create a new issue/PR to fix this issue for the other boards in one go, so we can keep things separate:
https://github.com/RIOT-OS/RIOT/blob/master/cpu/stm32/periph/adc_f0_g0_c0.c#L103-L106
https://github.com/RIOT-OS/RIOT/blob/master/cpu/stm32/periph/adc_f2.c#L112-L114
https://github.com/RIOT-OS/RIOT/blob/master/cpu/stm32/periph/adc_f3.c#L196-L199
https://github.com/RIOT-OS/RIOT/blob/master/cpu/stm32/periph/adc_f4_f7.c#L131-L134
https://github.com/RIOT-OS/RIOT/blob/master/cpu/stm32/periph/adc_wl.c#L121-L124


Better examples that don't rely on magic numbers:
https://github.com/RIOT-OS/RIOT/blob/master/cpu/stm32/periph/adc_f1.c#L146-L149
https://github.com/RIOT-OS/RIOT/blob/master/cpu/stm32/periph/adc_l0.c#L120-L126
https://github.com/RIOT-OS/RIOT/blob/master/cpu/stm32/periph/adc_l1.c#L149-L155